sdk/eventhub/event-hubs/src/eventHubConsumerClient.ts

```diff
@@ -188,22 +150,195 @@ export class EventHubConsumerClient {
+   *
+   * Use this overload if you want to read from all partitions and not coordinate with
+   * other subscribers.
+   *
+   * @param consumerGroupName The name of the consumer group from which you want to process events.
+   * @param onReceivedEvents Called when new events are received.
+   * @param options Options to handle additional events related to partitions (errors, 
+   *                opening, closing) as well as batch sizing.
    */
-  async getPartitionProperties(
-    partitionId: string,
-    options: GetPartitionPropertiesOptions = {}
-  ): Promise<PartitionProperties> {
-    return this._client.getPartitionProperties(partitionId, options);
+  subscribe(consumerGroupName: string, onReceivedEvents: OnReceivedEvents, options?: SubscriptionOptions): Subscription;   // #1
+  /**
+   * Subscribe to all messages from a subset of partitions.
+   *
+   * Use this overload if you want to read from a specific set of partitions and not coordinate with
+   * other subscribers.
+   *
+   * @param consumerGroupName The name of the consumer group from which you want to process events.
+   * @param onReceivedEvents Called when new events are received.
+   * @param partitionIds An array of partition ids to subscribe to.
+   * @param options Options to handle additional events related to partitions (errors, 
+   *                opening, closing) as well as batch sizing.
+   */  
+  subscribe(consumerGroupName: string, onReceivedEvents: OnReceivedEvents, partitionIds: string[], options?: SubscriptionOptions): Subscription;   // #2
+  /**
+   * Subscribes to multiple partitions.
+   *
+   * Use this overload if you want to coordinate with other subscribers using a `PartitionManager`
+   *
+   * @param consumerGroupName The name of the consumer group from which you want to process events.
+   * @param onReceivedEvents Called when new events are received.
+   *                         This is also a good place to update checkpoints as appropriate.
+   * @param partitionManager A partition manager that manages ownership information and checkpoint details.
+   * @param options Options to handle additional events related to partitions (errors, 
+   *                opening, closing) as well as batch sizing.
+   */
+  subscribe(
+    consumerGroupName: string,
+    onReceivedEvents: OnReceivedEvents,
+    partitionManager: PartitionManager,
+    options?: SubscriptionOptions
+  ): Subscription;      // #3
+  subscribe(
+    consumerGroupName1: string,
+    onReceivedEvents2: OnReceivedEvents,
+    optionsOrPartitionIdsOrPartitionManager3: SubscriptionOptions | undefined | string[] | PartitionManager,
+    possibleOptions4?: SubscriptionOptions
+  ): Subscription {
+    let eventProcessor: EventProcessor;
+
+    if (Array.isArray(optionsOrPartitionIdsOrPartitionManager3)) {
+      // #2: subscribe overload (read from specific partition IDs), don't coordinate
+      const partitionIds = optionsOrPartitionIdsOrPartitionManager3 as string[];
```

Comment:
Could this type be inferred as well given the Array.isArray check above?

Reply:
It totally could. I'm impressed with it. There's one more below as well that I got for free as well (isPartitionManager)